### PR TITLE
Unit test for CustomClusterController

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR do unit test for customcluster controller. Here is the Coverage
customcluster_controller.go:317:	reconcileProvision			100.0%
customcluster_controller.go:380:	reconcileDelete				95.2%
customcluster_controller.go:420:	deleteWorkerPods			63.6%
customcluster_controller.go:445:	deleteResource				60.0%
customcluster_controller.go:484:	ensureFinalizerAndOwnerRef		77.8%
customcluster_controller.go:516:	WorkerToCustomClusterMapFunc		85.7%
customcluster_controller.go:530:	CustomMachineToCustomClusterMapFunc	100.0%
customcluster_controller.go:548:	ClusterToCustomClusterMapFunc		85.7%
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
for the DCO failure: run `git rebase HEAD~2 --signoff` and then force push

**Comment by Xieql** [comment]
run `make gen` on your local corresponding branch, wait for the script to finish, and then push it again.

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,799 @@
...
+			Finalizers: []string{
+				"customcluster.cluster.kurator.dev",
+			},
+		},
+	}
+	return testCustomCluster
+}
+
+func generateCluster(clusterName string) *clusterv1.Cluster {
+	cluster := &clusterv1.Cluster{
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:110]:
```suggestion
	return  &clusterv1.Cluster{
```

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,799 @@
...
+			fields: fields{
+				Client: fake.NewClientBuilder().WithScheme(scheme.Scheme).
+					WithObjects(testPod, testCustomMachine, testCustomCluster).Build(),
+			},
+			args: args{
+				o: testPod,
+			},
+		},
+		{
+			name:      "CustomManchine test",
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:308]: nit: s/CustomManchine/CustomMachine

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,799 @@
...
+		o client.Object
+	}
+	tests := []struct {
+		name      string
+		wantError bool
+		fields    fields
+		args      args
+	}{
+		{
+			name:      "Not CustomMachine Error",
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:297]: nit: recommend all using small case

**Review by hzxuzhonghu — APPROVED** [review]
good work

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,799 @@
...
+				Client:    tt.fields.Client,
+				APIReader: tt.fields.APIReader,
+				Scheme:    tt.fields.Scheme,
+			}
+			err1 := r.deleteWorkerPods(tt.args.ctx, tt.args.customCluster)
+			assert.Empty(t, err1)
+
+			patches := gomonkey.ApplyPrivateMethod(reflect.TypeOf(r), "ensureWorkerPodDeleted",
+				func(_ *CustomClusterController, ctx context.Context, customCluster *v1alpha1.CustomCluster, manageAction customClusterManageAction) error {
+					err := errors.New("failede to delete Pod")
```
Comment (Xieql) [inline pkg/cluster-operator/customcluster_controller_test.go:267]: nit: s/failede /failed

**Comment by Xieql** [comment]
please set a new PR name that includes more information, such as "Unit test for CustomClusterController"

**Review by hzxuzhonghu — COMMENTED** [review]
/lgtm

**Pull request merged**